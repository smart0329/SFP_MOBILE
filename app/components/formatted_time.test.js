// Copyright (c) 2019-present Smart.Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {render} from '@testing-library/react-native';
import {IntlProvider} from 'react-intl';
import moment from 'moment-timezone';

import FormattedTime from './formatted_time';

describe('FormattedTime', () => {
    const baseProps = {
        value: 1548788533405,
        timeZone: 'UTC',
        hour12: true,
    };

    it('should render correctly', () => {
        console.error = jest.fn();

        let wrapper = renderWithIntl(
            <FormattedTime {...baseProps}/>,
        );

        expect(wrapper.baseElement).toMatchSnapshot();
        expect(wrapper.getByText('7:02 PM')).toBeTruthy();

        wrapper = renderWithIntl(
            <FormattedTime
                {...baseProps}
                hour12={false}
            />,
        );

        expect(wrapper.getByText('19:02')).toBeTruthy();
    });

    it('should support localization', () => {
        moment.locale('es');
        let wrapper = renderWithIntl(
            <FormattedTime {...baseProps}/>,
            'es',
        );

        expect(wrapper.getByText('7:02 PM')).toBeTruthy();

        moment.locale('ko');
        wrapper = renderWithIntl(
            <FormattedTime {...baseProps}/>,
            'ko',
        );

        expect(wrapper.getByText('오후 7:02')).toBeTruthy();

        wrapper = renderWithIntl(
            <FormattedTime
                {...baseProps}
                hour12={false}
            />,
            'ko',
        );

        expect(wrapper.getByText('19:02')).toBeTruthy();
    });

    it('should fallback to default short format for unsupported locale of react-intl ', () => {
        moment.locale('es');
        let wrapper = renderWithIntl(
            <FormattedTime
                {...baseProps}
                timeZone='NZ-CHAT'
            />,
            'es',
        );

        expect(wrapper.getByText('8:47 AM')).toBeTruthy();

        wrapper = renderWithIntl(
            <FormattedTime
                {...baseProps}
                timeZone='NZ-CHAT'
                hour12={false}
            />,
            'es',
        );

        expect(wrapper.getByText('8:47')).toBeTruthy();
    });
});

function renderWithIntl(component, locale = 'en') {
    return render(<IntlProvider locale={locale}>{component}</IntlProvider>);
}
