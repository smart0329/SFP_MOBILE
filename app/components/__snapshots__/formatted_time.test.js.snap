// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormattedTime should render correctly 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text>
      7:02 PM
    </Text>
  </View>
</View>
`;
